### ISSUE TITLE
Other Graph Operators.

### ISSUE THREAD

**Issue by dlee138** [Statistical-Connectomics-Sp15/intro#31]
Last class we talked about adding two different graphs, so aren't other operations such as graph subtraction and graph products possible as well?  For example, if we were to select particular features of shape A and shape B and added them to get a new shape (lets call it shape C), and then we subtract the selected features of shape A from shape C, would we end up with the original shape B?
-David

**Comment by imichaelnorris** [comment]
If you add the edges from graphs A and B to a new graph C, then remove the edges only from A, you will be left with the graph  B.  If there are edges in both A and B, and you remove A's edges from C, then you will be left with the graph B minus any edges that are in A as well as B.

Graphs don't really have a concept of shape.  There's only nodes and edges.  Of course, if the graph is representing something physical like a brain then there will be an an underlying structure to the graph.

**Comment by mrjiaruiwang** [comment]
Yes that's very interesting. I would guess that something similar to multiplication or even multivariate convolution may be at play when the brain develops. From a mathematical/computational point of view, the brain seems to be extremely complicated, but maybe simpler models can be built where the mathematics is simple, but the phenotypical output may not necessarily be simple.

**Comment by jovo** [comment]
kronecker graphs get at multiplication for growth a little bit.

in general, any mathemtical operation you know and love doesn't make sense
on graphs lacking labels.
if you assume the graphs have labels, and you know them, and they are
weighted,
then additional and subtraction make sense, and maybe element wise
multiplication and division,
but i'm not sure what matrix multiply and division would mean.

On Tuesday, February 3, 2015, Jerry Wang <notifications@github.com> wrote:

>  Yes that's very interesting. I would guess that something similar to
> multiplication or even multivariate convolution may be at play when the
> brain develops. From a mathematical/computational point of view, the brain
> seems to be extremely complicated, but maybe simpler models can be built
> where the mathematics is simple, but the phenotypical output may not
> necessarily be simple.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Statistical-Connectomics-Sp15/intro/issues/31#issuecomment-72786616>
> .
>


-- 
the glass is all full:  half water, half air.
openconnecto.me, we're hiring!
<https://docs.google.com/document/d/14SApYAzxF0Ddqg2ZCEwjmz3ht2TDhDmxyZI2ZP82_0U/edit?usp=sharing>
, jovo.me, my calendar
<https://www.google.com/calendar/embed?src=joshuav%40gmail.com&ctz=America/New_York>
